packages/protocol/contracts/tokenvault/BridgedERC1155.sol

```diff
@@ -122,6 +122,27 @@ contract BridgedERC1155 is EssentialContract, ERC1155Upgradeable {
         return LibBridgedToken.buildSymbol(__symbol);
     }
 
+    /// @notice Gets the source token and source chain ID being bridged.
+    /// @return The source token's address.
+    /// @return The source token's chain ID.
+    function source() public view returns (address, uint256) {
+        return (srcToken, srcChainId);
+    }
+
+    /// @notice Returns the token URI.
+    /// @param _tokenId The token id.
+    /// @return The token URI following EIP-681.
+    function uri(uint256 _tokenId) public view override returns (string memory) {
```

Comment:
I understand this differently: missing _tokenId in the URL seems fine based on the 1155 standard, as long as the app knows the value, it can replace {id}.

Reply:
But in our contract there is nothing which set the _uri in `ERC1155Upgradeable`, so if we leave it as it, it will always return "" (empty string).

Reply:
line 54 does it `__ERC1155_init(LibBridgedToken.buildURI(_srcToken, _srcChainId));`

Inside ERC1155Upgradeable.sol, there is:

```
  function __ERC1155_init(string memory uri_) internal onlyInitializing {
        __ERC1155_init_unchained(uri_);
    }

    function __ERC1155_init_unchained(string memory uri_) internal onlyInitializing {
        _setURI(uri_);
    }
    
    function _setURI(string memory newuri) internal virtual {
        _uri = newuri;
    }
    
     function uri(uint256) public view virtual override returns (string memory) {
        return _uri;
    }
```

Reply:
Right, i missed that buildURI, but still, it is incorrect for ERC1155.

To conform the erc1155, you should return literally `{id}.json`.

Please see it here, i did this contract, query the uri() function:
https://polygonscan.com/token/0x88883145049dec03c00cb7708cbc938058802520#readContract

Reply:
But i'm not sure if we shall add that at this point then.. Because we using another standard to build this URI.. If we add `{id}.json` -> it might have no effect, since the dApps could easily query this information cross-chain.